tests/core/stream_muxer/test_yamux.py

```diff
@@ -0,0 +1,195 @@
+import pytest
+import trio
+from trio.testing import (
+    memory_stream_pair,
+)
+
+from libp2p.crypto.ed25519 import (
+    create_new_key_pair,
+)
+from libp2p.peer.id import (
+    ID,
+)
+from libp2p.security.insecure.transport import (
+    InsecureTransport,
+)
+from libp2p.stream_muxer.yamux.yamux import (
+    MuxedStreamError,
+    Yamux,
+    YamuxStream,
+)
+
+
+class TrioStreamAdapter:
+    def __init__(self, send_stream, receive_stream):
+        self.send_stream = send_stream
+        self.receive_stream = receive_stream
+
+    async def write(self, data):
+        print(f"Writing {len(data)} bytes")
```

Comment:
`print` statements are fine during development, but we'd prefer to use logging in production. That way the output doesn't clutter up CI logs, but can be turned on when running pytest locally.

Reply:
Totally agree on using `logging` over `print` for production readiness. I’ve replaced the `print` statements in `test_yamux.py` with `logging.debug`.